PSKoans/Koans/Foundations/AboutHashtables.Koans.ps1

```diff
@@ -96,45 +101,58 @@ Describe 'Hashtables' {
             $HashtableTwo['Calories'] = 250
             $HashtableTwo['Contents'] = 'Chips'
 
-            $HashtableOne['Meal Type'] | Should -Be '__'
+            '____' | Should -Be $HashtableOne['Meal Type']
             __ | Should -Be $HashtableOne['Calories']
 
-            $HashtableTwo['Contents'] | Should -Be '__'
+            '____' | Should -Be $HashtableTwo['Contents']
         }
 
         It 'allows you to retrieve a list of keys or values' {
-            $Hashtable = @{One = 1; Two = 2; Three = 3; Four = 4 }
+            $Hashtable = @{ One = '1'; Two = '2'; Three = '3'; Four = '4' }
 
-            $Hashtable.Keys | Should -Be @('__', '__', '__', '__')
-            $Hashtable.Values | Should -Be @( )
+            @( 'One', '____', '____', '____' ) | Should -BeIn $Hashtable.Keys
+            @( '__', '__', '__', '__' ) | Should -BeIn $Hashtable.Values
         }
 
         It 'is not ordered' {
             # Hashtables are ordered by hashing their keys for extremely quick lookups.
-            $Hashtable = @{One = 1; Two = 2; Three = 3; Four = 4 }
+            $Hashtable = @{ One = 1; ____ = 2; Three = 3; Four = __ }
 
-            # You will find your key/value pairs are often not at all in the order you entered them.
-            # This _probably_ won't pass.
-            $Hashtable.Keys | Should -Be @('One', 'Two', 'Three', 'Four')
-            $Hashtable.Values | Should -Be @(1, 2, 3, 4)
+            <#
+                You will find your key/value pairs are often not at all in the order you entered them.
+                Trying to predict the order of the pairs like this can be very difficult.
+                The following assertions would be highly likely to fail.
 
-            # The order can and will change again, as well, if the collection is changed.
-            $Hashtable['Five'] = 5
+                    $Hashtable.Keys | Should -Be @('One', 'Two', 'Three', 'Four')
+                    $Hashtable.Values | Should -Be @(1, 2, 3, 4)
 
-            $Hashtable.Keys | Should -Be @('One', 'Two', 'Three', 'Four', 'Five')
-            $Hashtable.Values | Should -Be @(1, 2, 3, 4, 5)
+                The order can and will change again, as well, if the collection is altered:
+
+                    $Hashtable['Five'] = 5
+            #>
+
+            $Hashtable.Keys | Should -Not -Be @('One', 'Two', 'Three', 'Four', 'Five')
+            $Hashtable.Values | Should -Not -Be @(1, 2, 3, 4, 5)
+
+            $Hashtable.Keys | Should -BeIn @('____', 'Two', '____', 'Four', 'Five')
```

Comment:
This looks good!

Reply:
I think the chances are sufficiently low, but if we see it become an issue we can relegate those two to comments instead I suppose. 🙂